lib/mllp/sender.ex

```diff
@@ -293,54 +275,78 @@ defmodule MLLP.Sender do
 
     payload = message |> MLLP.Envelope.wrap_message()
 
-    state.tcp.send(state.socket, payload)
-    |> case do
+    case state.tcp.send(state.socket, payload) do
       :ok ->
         reply = {:ok, :sent}
 
         telemetry(:received, %{response: reply}, state)
         {:reply, reply, state}
 
       {:error, reason} ->
-        telemetry(
-          :status,
-          %{status: :disconnected, error: reason, context: "send message failure"},
-          state
-        )
-
-        new_state = maintain_reconnect_timer(state)
-        reply = {:error, :send_message_failure}
-        {:reply, reply, new_state}
+        handle_send_error(state, reason)
     end
   end
 
-  def handle_info({:tcp_closed, _socket}, state) do
-    telemetry(:status, %{status: :disconnected, error: :tcp_closed}, state)
-    {:noreply, maintain_reconnect_timer(state)}
+  def handle_info(:connect_timeout, state) do
+    new_state =
+      case Process.info(state.socket) do
```

Comment:
This is defensive and generally goes against the erlang way ™️ , however, being defensive here might serve us well. I also need to double check that this method of checking if the socket still  exists is  reliable. Further, per  other changes, this might not be needed.